pkg/util/push/otel.go

```diff
@@ -150,10 +152,49 @@ func OTLPHandler(
 		)
 
 		req.Timeseries = metrics
+
+		if enableOtelMetadataStorage {
+			metadata := otelMetricsToMetadata(otlpReq.Metrics())
+			req.Metadata = metadata
+		}
+
 		return body, nil
 	})
 }
 
+func otelMetricsToMetadata(md pmetric.Metrics) []*mimirpb.MetricMetadata {
+	resourceMetricsSlice := md.ResourceMetrics()
+
+	metadataLength := 0
+	for i := 0; i < resourceMetricsSlice.Len(); i++ {
+		metadataLength += resourceMetricsSlice.At(i).ScopeMetrics().Len()
+	}
+	var metadata = make([]*mimirpb.MetricMetadata, metadataLength)
+	var metaDataPos = 0
+	for i := 0; i < resourceMetricsSlice.Len(); i++ {
+		resourceMetrics := resourceMetricsSlice.At(i)
+		scopeMetricsSlice := resourceMetrics.ScopeMetrics()
+
+		for j := 0; j < scopeMetricsSlice.Len(); j++ {
+			scopeMetrics := scopeMetricsSlice.At(j)
+			for k := 0; k < scopeMetrics.Metrics().Len(); k++ {
+				metric := scopeMetrics.Metrics().At(k)
+				entry := mimirpb.MetricMetadata{
+					Type:             mimirpb.MetricMetadata_MetricType(metric.Type()),
```

Comment:
I'm confused here. `metric.Type()` returns this: https://pkg.go.dev/go.opentelemetry.io/collector/pdata/pmetric#MetricType

While https://github.com/grafana/mimir/blob/main/pkg/mimirpb/mimir.pb.go#L53-L64 is a different thing. Specifically OTLP has "Sum" while Mimir has "COUNTER". How are we mapping them together?

Reply:
Yeah I will look into how is this handled in https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/prometheusexporter

Reply:
Added the function for translating OTEL metric types to Prometheus metric types.